src/sentry/models/projectcodeowners.py

```diff
@@ -118,3 +119,32 @@ def update_schema(self, raw: str | None = None) -> None:
                 self.save()
         except ValidationError:
             return
+
+
+def process_resource_change(instance, change, **kwargs):
+    from sentry.models import GroupOwner, ProjectOwnership
+
+    cache.set(
+        ProjectCodeOwners.get_cache_key(instance.project_id),
+        None,
+        READ_CACHE_DURATION,
+    )
+    ownership = ProjectOwnership.get_ownership_cached(instance.project_id)
+    if not ownership:
+        ownership = ProjectOwnership(project_id=instance.project_id)
+
+    autoassignment_types = ProjectOwnership._get_autoassignment_types(ownership)
+    GroupOwner.invalidate_autoassigned_owner_cache(instance.project_id, autoassignment_types)
```

Comment:
One thought I had here is that this might slow down response times when saving `ProjectCodeOwners`. I ran a quick query, and most projects have 1000 or fewer groups in the last hour, so I think that this is fine.

One option here if we do find this to be a problem is that we could run this invalidation logic in a task instead.